server/src/subscriber/subscriber.controller.ts

```diff
@@ -26,6 +30,13 @@ export class SubscriberController {
       return;
     }
 
+    if(!validateSubscriptions(request.body.subscriptions)) {
```

Comment:
This validation feels business logic-y so probably best to put it in the service. The controller calls `create` on the service, and that function handles validation.